src/core/logging.js

```diff
@@ -34,10 +35,21 @@ export function registerLoggingCallbacks( obj ) {
 }
 
 export function runLoggingCallbacks( key, args ) {
-	var i, l, callbacks;
-
-	callbacks = config.callbacks[ key ];
-	for ( i = 0, l = callbacks.length; i < l; i++ ) {
-		callbacks[ i ]( args );
+	var callbacks = config.callbacks[ key ];
+
+	// Handling 'log' callbacks separately. Unlike the other callbacks,
+	// the log callback is not controlled by the processing queue,
+	// but rather used by asserts. Hence to promisfy the 'log' callback
+	// would mean promisfying each step of a test
+	if ( key === "log" ) {
```

Comment:
I think this is a reasonable tradeoff to make. We should add a note to the docs however that this callback must be synchronous.

Reply:
Added notes to the doc for QUnit.log and QUnit.on